protob/pb2py

```diff
@@ -0,0 +1,357 @@
+            yield ""
+            yield from self.create_init_method(fields)
+
+        if not fields and not msg_id:
+            yield "    pass"
+
+    def process_enum(self, enum):
+        logging.debug("Processing enum %s", enum.name)
+
+        for value in enum.value:
+            # Remove type name from the beginning of the constant
+            # For example "PinMatrixRequestType_Current" -> "Current"
+            enum_prefix = enum.name
+            name = value.name
+            name = remove_from_start(name, "%s_" % enum_prefix)
+
+            # If type ends with *Type, but constant use type name without *Type, remove it too :)
+            # For example "ButtonRequestType & ButtonRequest_Other" => "Other"
+            if enum_prefix.endswith("Type"):
+                enum_prefix, _ = enum_prefix.rsplit("Type", 1)
+                name = remove_from_start(name, "%s_" % enum_prefix)
+
+            yield "%s = %s" % (name, value.number)
+
+    def process_messages(self, messages):
+        for message in sorted(messages, key=lambda m: m.name):
+            self.write_to_file(message.name, self.process_message(message))
+            self.write_to_index(self.create_message_import(message.name))
+
+    def process_enums(self, enums):
+        for enum in sorted(enums, key=lambda e: e.name):
+            self.write_to_file(enum.name, self.process_enum(enum))
+            self.write_to_index("from . import %s" % enum.name)
+
+    def write_to_file(self, name, out):
+        # Write generated sourcecode to given file
+        logging.debug("Writing file %s.py" % name)
+        with open(os.path.join(self.out_dir, "%s.py" % name), "w") as f:
+            f.write(AUTO_HEADER)
+            for line in out:
+                f.write(line + "\n")
+
+    def write_to_index(self, statement):
+        if self.module_index:
+            self.module_index.write(statement + "\n")
+
+    def write_classes(self, out_dir, module_index=None):
+        self.out_dir = out_dir
+        self.module_index = module_index
+        self.process_messages(self.messages)
+        self.process_enums(self.enums)
+
+
+if __name__ == "__main__":
+    logging.basicConfig(level=logging.DEBUG)
+
+    parser = argparse.ArgumentParser()
+    # fmt: off
+    parser.add_argument('proto', help="Protobuf definition file")
+    parser.add_argument('out_dir', help="Directory for generated source code")
+    parser.add_argument('-p', '--package', default="trezor", help="Name of protobuf package to extract. Use '-p all' to process all packages")
+    parser.add_argument('-P', '--protobuf-module', default="protobuf", help="Name of protobuf module")
+    parser.add_argument('-l', '--module-index', help="Generate list of modules")
+    parser.add_argument('--message-type', default="MessageType", help="Name of enum with message IDs")
+    parser.add_argument("--protobuf-default-include", default="/usr/include", help="Location of protobuf's default .proto files.")
```

Comment:
Is `-I/usr/include` actually necessary? I've never had any issues without it.

Reply:
turns out it is. our protobufs require `descriptor.proto` which lives there by default

Reply:
Which version of `protoc`? Imports fine without, for me at least.

Reply:
libprotoc 2.6.1. That's rather old, perhaps the new versions include this by default; nevertheless, i do actually need it :)